### ISSUE TITLE
Some Fields are Placed Randomly in `modrinth.index.json` When Exporting

### ISSUE THREAD

**Issue by UltimatChamp** [modrinth/code#2510]
### Please confirm the following.

- [x] I checked the [existing issues](https://github.com/modrinth/code/issues) for duplicate problems
- [x] I have tried resolving the issue using the [support portal](https://support.modrinth.com)
- [x] I have ensured my Modrinth App installation is up to date

### What version of the Modrinth App are you using?

0.8.8

### Describe the bug

<img width="450px" src="https://github.com/user-attachments/assets/16947f16-7e57-4a46-b13a-5d79bed06ff1" />

It looks like the `sha1` and `sha512` fields are placed, randomly, while exporting a modpack. Same is the case for `client`-`server` and the `minecraft`-`loader` fields. The image shows a diff between two different, simultaneous exports.

**Comment by Norbiros** [comment]
I don't think this is really an issue. We use hashmaps to store envs/hashes, which means that fields may be unordered when serialized. JSON objects don't require a specific order, and there are probably many other places that also have "random" order

**Comment by UltimatChamp** [comment]
Yeah, but it's pretty annoying if you upload-n-update files to GitHub.
I'm sure that at some point, it did not happen in Modrinth App.

**Comment by Norbiros** [comment]
We could sort it here or use a btree, but there may be other places with a 'random' order. Have you noticed any others?

**Comment by UltimatChamp** [comment]
These
> It looks like the `sha1` and `sha512` fields are placed, randomly, while exporting a modpack. Same is the case for `client`-`server` and the `minecraft`-`loader` fields. The image shows a diff between two different, simultaneous exports.

Oh, yeah! The project files are also randomly placed. So, maybe it's time to fix #1115
